### PR TITLE
Bump compatiblity check for SDK 0.50 and IBC-Go v8

### PR DESCRIPTION
**Pull request by romac**
Closes: #3745

## Description

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [x] Linked to GitHub issue.
- [x] Updated code comments and documentation (e.g., `docs/`).
- [x] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/relayer/src/chain/cosmos/compatibility.rs [hunk 1]
```diff
@@ -20,7 +20,7 @@ const SDK_MODULE_VERSION_REQ: &str = ">=0.44, <0.48";
 /// # Note: Should be consistent with [features] guide page.
 ///
 /// [features]: https://hermes.informal.systems/advanced/features.html
-const IBC_GO_MODULE_VERSION_REQ: &str = ">=1.1, <=7";
+const IBC_GO_MODULE_VERSION_REQ: &str = ">=1.1, <9";
```
Comment (ljoss17) [inline crates/relayer/src/chain/cosmos/compatibility.rs:23]: The 1.1 minimum version seems low, should we increase it to v4 at least?
I don't recall seeing any chains using lower than v4 and in theory ibc-go v6.1 and v6.2 reached end of life the 9th of December 2023 (but there are still many chains on v4)

**Review by ljoss17 — APPROVED** [review]
Looks good to me. Just a small question

File: crates/relayer/src/chain/cosmos/compatibility.rs (hunk 1 above)
Comment (romac) [reply crates/relayer/src/chain/cosmos/compatibility.rs:23]: Good point, yeah let's raise it to v4 for now and v7 in the next Hermes release.

File: guide/src/advanced/features.md [hunk 2]
```diff
@@ -4,8 +4,8 @@ This section includes a summary of the supported and planned features. It also i
 
 > **Cosmos SDK & IBC compatibility:**
 > Hermes supports Cosmos SDK chains implementing the [IBC protocol v1][ibcv1-proto] protocol specification.
-> Cosmos SDK versions `0.44.0` through `0.47.x` are officially supported.
-> IBC-go versions `1.1.*` thorough `7.*` are officially supported.
+> Cosmos SDK versions `0.44.0` through `0.50.x` are officially supported.
+> IBC-go versions `1.1.*` through `8.*` are officially supported.
```
Comment (ljoss17) [inline guide/src/advanced/features.md:8]:
```suggestion
> IBC-go versions `4.1.*` through `8.*` are officially supported.
```

**Review by ljoss17 — APPROVED** [review]
Looks good to me, only missing the update in the guide

File: guide/src/advanced/features.md [hunk 3]
```diff
@@ -4,8 +4,8 @@ This section includes a summary of the supported and planned features. It also i
 
 > **Cosmos SDK & IBC compatibility:**
 > Hermes supports Cosmos SDK chains implementing the [IBC protocol v1][ibcv1-proto] protocol specification.
-> Cosmos SDK versions `0.44.0` through `0.47.x` are officially supported.
-> IBC-go versions `1.1.*` thorough `7.*` are officially supported.
+> Cosmos SDK versions `0.44.0` through `0.50.x` are officially supported.
```
Comment (ljoss17) [inline guide/src/advanced/features.md:7]:
```suggestion
> Cosmos SDK versions `0.45.0` through `0.50.x` are officially supported.
```
Sorry I missed this during the previous review

**Pull request merged**